I have only an optional suggestion, though pylint is not happy


************* Module noxfile
noxfile.py:1:0: R0401: Cyclic import (src.frequenz.sdk.timeseries -> src.frequenz.sdk.timeseries._moving_window) (cyclic-import)
noxfile.py:1:0: R0401: Cyclic import (src.frequenz.sdk.timeseries -> src.frequenz.sdk.timeseries._moving_window -> src.frequenz.sdk.timeseries._resampling) (cyclic-import)
noxfile.py:1:0: R0401: Cyclic import (src.frequenz.sdk.timeseries -> src.frequenz.sdk.timeseries._resampling) (cyclic-import)
noxfile.py:1:0: R0401: Cyclic import (src.frequenz.sdk.timeseries -> src.frequenz.sdk.timeseries._moving_window -> src.frequenz.sdk.timeseries._ringbuffer -> src.frequenz.sdk.timeseries._ringbuffer.serialization -> src.frequenz.sdk.timeseries._ringbuffer.buffer) (cyclic-import)
noxfile.py:1:0: R0401: Cyclic import (src.frequenz.sdk.timeseries -> src.frequenz.sdk.timeseries._moving_window -> src.frequenz.sdk.timeseries._ringbuffer -> src.frequenz.sdk.timeseries._ringbuffer.buffer) (cyclic-import)

--- a/src/frequenz/sdk/timeseries/_moving_window.py
+++ b/src/frequenz/sdk/timeseries/_moving_window.py
@@ -17,8 +17,7 @@ from frequenz.channels import Broadcast, Receiver, Sender
 from numpy.typing import ArrayLike
 
 from .._internal.asyncio import cancel_and_await
-from . import Sample
-from ._base_types import UNIX_EPOCH
+from ._base_types import UNIX_EPOCH, Sample
 from ._resampling import Resampler, ResamplerConfig
 from ._ringbuffer import OrderedRingBuffer
 

--- a/src/frequenz/sdk/timeseries/_resampling.py
+++ b/src/frequenz/sdk/timeseries/_resampling.py
@@ -16,8 +16,7 @@ from datetime import datetime, timedelta, timezone
 from typing import AsyncIterator, Callable, Coroutine, Optional, Sequence
 
 from .._internal.asyncio import cancel_and_await
-from . import Sample
-from ._base_types import UNIX_EPOCH
+from ._base_types import UNIX_EPOCH, Sample
 
 _logger = logging.getLogger(__name__)
 

--- a/src/frequenz/sdk/timeseries/_ringbuffer/buffer.py
+++ b/src/frequenz/sdk/timeseries/_ringbuffer/buffer.py
@@ -14,8 +14,7 @@ from typing import Generic, List, SupportsFloat, SupportsIndex, TypeVar, overloa
 import numpy as np
 import numpy.typing as npt
 
-from .. import Sample
-from .._base_types import UNIX_EPOCH
+from .._base_types import UNIX_EPOCH, Sample
 
 FloatArray = TypeVar("FloatArray", List[float], npt.NDArray[np.float64])
 

